### PR TITLE
feat: [CDS-54470]: Delete previous Perpetual Task if new Perpetual Task is successful  instance sync V2

### PR DESCRIPTION
**Pull request by namantalaycha10**
## Describe your changes
Delete previous Perpetual Task if new Perpetual Task is successful executed first time for instance sync V2

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48648)

### CODE REVIEW

**Comment by namantalaycha10** [comment]
trigger uts

File: 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java [hunk 1]
```diff
@@ -167,7 +169,8 @@ public void processInstanceSyncForNewDeployment(DeploymentEvent deploymentEvent)
 
           InstanceSyncPerpetualTaskInfoDTO instanceSyncPerpetualTaskInfoDTO;
 
-          if (instanceSyncPerpetualTaskService.isInstanceSyncV2Enabled()) {
+          if (instanceSyncPerpetualTaskService.isInstanceSyncV2Enabled()
+              && isSupportedDeploymentType(deploymentSummaryDTO.getDeploymentInfoDTO().getType())) {
```
Comment (abosii-harness) [inline 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java:173]: I don't think this is required, logic around if instance sync v2 is enabled should be based on AbstractInstanceSyncHandler, for now we can keep the current logic for testing purpose

**Review by abosii-harness — CHANGES_REQUESTED** [review]
Please review logic at io/harness/service/instancesync/InstanceSyncServiceImpl.java:298 when connector not found we're ending up in deleting v1 pt and not v2. As well null check is missing

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
trigger uts

File: 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java [hunk 2]
```diff
@@ -203,6 +205,10 @@ public void processInstanceSyncForNewDeployment(DeploymentEvent deploymentEvent)
     }
   }
 
+  private boolean isSupportedDeploymentType(String DeploymentType) {
```
Comment (abosii-harness) [inline 126-instance/src/main/java/io/harness/service/instancesync/InstanceSyncServiceImpl.java:208]: Not needed

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
trigger ut0

**Review by abosii-harness — APPROVED** [review]

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by namantalaycha10** [comment]
trigger review

**Pull request merged**